core/market/src/db/dao/proposal.rs

```diff
@@ -146,16 +166,12 @@ pub(super) fn has_counter_proposal(conn: &ConnType, proposal_id: &ProposalId) ->
     Ok(proposal.is_some())
 }
 
-pub(super) fn set_proposal_accepted(conn: &ConnType, proposal_id: &ProposalId) -> DbResult<()> {
-    // TODO: Check if we can do transition to this state.
-    update_proposal_state(conn, proposal_id, ProposalState::Accepted)
-}
-
```

Comment:
There was todo, that we should check state before transition. Now by making generic function you removed possibility to check this state in future

Reply:
I've reparsed the todo and moved it below and above in error struct